x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -844,38 +835,36 @@ private void preAnalyzeMainIndices(
         );
-        // TODO: This is not yet index specific, but that will not matter as soon as #136804 is dealt with
-        if (executionInfo.clusterAliases().isEmpty()) {
-            // return empty resolution if the expression is pure CCS and resolved no remote clusters (like no-such-cluster*:index)
-            listener.onResponse(
-                result.withIndices(indexPattern, IndexResolution.valid(new EsIndex(indexPattern.indexPattern(), Map.of(), Map.of())))
-            );
-        } else {
-            indexResolver.resolveAsMergedMappingAndRetrieveMinimumVersion(
-                indexPattern.indexPattern(),
-                result.fieldNames,
-                // Maybe if no indices are returned, retry without index mode and provide a clearer error message.
-                switch (indexMode) {
-                    case IndexMode.TIME_SERIES -> {
-                        var indexModeFilter = new TermQueryBuilder(IndexModeFieldMapper.NAME, IndexMode.TIME_SERIES.getName());
-                        yield requestFilter != null
-                            ? new BoolQueryBuilder().filter(requestFilter).filter(indexModeFilter)
-                            : indexModeFilter;
-                    }
-                    default -> requestFilter;
-                },
-                indexMode == IndexMode.TIME_SERIES,
-                preAnalysis.useAggregateMetricDoubleWhenNotSupported(),
-                preAnalysis.useDenseVectorWhenNotSupported(),
-                listener.delegateFailureAndWrap((l, indexResolution) -> {
-                    EsqlCCSUtils.updateExecutionInfoWithUnavailableClusters(executionInfo, indexResolution.inner().failures());
-                    l.onResponse(
-                        result.withIndices(indexPattern, indexResolution.inner())
-                            .withMinimumTransportVersion(indexResolution.minimumVersion())
-                    );
-                })
-            );
-        }
+        indexResolver.resolve(
+            indexPattern.indexPattern(),
+            result.fieldNames,
+            // Maybe if no indices are returned, retry without index mode and provide a clearer error message.
+            switch (indexMode) {
+                case IndexMode.TIME_SERIES -> {
+                    var indexModeFilter = new TermQueryBuilder(IndexModeFieldMapper.NAME, IndexMode.TIME_SERIES.getName());
+                    yield requestFilter != null ? new BoolQueryBuilder().filter(requestFilter).filter(indexModeFilter) : indexModeFilter;
+                }
+                default -> requestFilter;
+            },
+            indexMode == IndexMode.TIME_SERIES,
+            preAnalysis.useAggregateMetricDoubleWhenNotSupported(),
+            preAnalysis.useDenseVectorWhenNotSupported(),
+            listener.delegateFailureAndWrap((l, indexResolution) -> {
+                EsqlCCSUtils.initCrossClusterInfo(indicesExpressionGrouper, verifier.licenseState(), indexPattern, executionInfo);
```

Comment:
Here the state will be populated from field caps `resolvedTo` data structure if that is available. 
We will keep existing approach for responses from older clusters that do not have this information available.